Split the Matroska specifications in 3 different documents

robUx4 opened the pull request:
- Matroska specifications
- Codec specifications
- Tags specifications

Fixes https://github.com/Matroska-Org/matroska-specification/issues/143

robUx4 commented:
Oops duplicate of #143

robUx4 commented:
And I meant to close the double issue...

robUx4 commented:
Pending approval at the IETF99 meeting.

ablwr commented:
What did IETF99 have to say about this?

robUx4 commented:
My understanding was that it's OK to split. Did anyone else attending understand the same thing ? @dericed @retokromer @mbunkus ?

retokromer commented:
Yes, I did so.

robUx4 commented:
From https://datatracker.ietf.org/meeting/99/materials/minutes-99-cellar

```Tim (as chair): Is the WG ok with moving codec mappings and tags to separate specs?
Jonathan: Codec registrations feels like RTP payload formats. Will IANA take over this?
Steve: Not sure how IANA works. (Tim offered links.) Need to explore that as a possible direction.```

mbunkus commented on index_codec.md:
```diff
@@ -3,46 +3,17 @@
 # Introduction
 
-Matroska aims to become THE standard of multimedia container formats. It was derived from a project called [MCF](http://mukoli.free.fr/mcf/mcf.html), but differentiates from it significantly because it is based on [EBML](https://github.com/Matroska-Org/ebml-specification/blob/master/specification.markdown) (Extensible Binary Meta Language), a binary derivative of XML. EBML enables significant adv ...
+Matroska aims to become THE standard of multimedia container formats. It stores interleaved and timestamped audio/video/subtitle data using various codec. To interpret the codec data, a mapping between the way the data are stored in Matroska and how they are understood by such codec is necessary.
```
"…using various codec" ← shouldn't that be plural, "codecs"?

mbunkus replied:
…and here we need singular: "…a mapping between the way the data are stored…" → "…is stored…".

Additionally: "…and how they are understood by such codec is necessary" This is difficult to understand. My guess is your "they" refers to the "the data", which are inherently singular in English. Meaning the whole sentence should probably be:

"To interpret the codec data, a mapping between the way the data is stored in Matroska and how it is understood by such a codec is necessary."

mbunkus commented on index_codec.md:
```diff
@@ -3,46 +3,17 @@
-- Menus (like DVDs have)
-
-Matroska is an open standards project. This means for personal use it is absolutely free to use and that the technical specifications describing the bitstream are open to everybody, even to companies that would like to support it in their products.
+This document intends to define this mapping for many commonly used codec in Matroska.
```
"codecs", plural. There's nothing in English that says that "codec" must be singular. Basically it's an abbreviation for "coder/decoder", and both are countable (meaning they have regular plural forms: coders, decoders). Hence it stands to reason that their shorter, fused variant also follows regular rules: codecs.

mbunkus commented on index_codec.md:
```diff
@@ -3,46 +3,17 @@
-The table found below is now generated from the "source" of the Matroska specification. This [XML file](https://github.com/Matroska-Org/foundation-source/blob/master/spectool/specdata.xml) is also used to generate the semantic data used in libmatroska and libmatroska2\. We encourage anyone to use and monitor its changes so your code is spec-proof and always up to date.
-
-Note that versions 1, 2 and 3 have been finalized. Version 4 is currently work in progress. There MAY be further additions to v4.
+This document is a work-in-progress specification defining the Matroska file format as part of the [IETF Cellar working group](https://datatracker.ietf.org/wg/cellar/charter/). It uses basic elements and concept already defined in the Matroska specifications defined by this workgroup.
```
Plural, please: "concepts"

mbunkus commented on index_codec.md:
```diff
@@ -1,6 +1,42 @@
+
+.# Abstract
+
+This document defines the Matroska codec mappings, including the codec ID, layout of data in a `Block` and in an optinal `CodecPrivate`.
```
"optinal" → "optional"

mbunkus commented on index_matroska.md:
```diff
@@ -1,6 +1,42 @@
+
+.# Abstract
+
+This document defines the Matroska audiovisual container, including definitions of its structural Elements, as well as its terminology, vocabulary, and application.
```
Either backtick "Elements" or use lower-case.

mbunkus commented on index_tags.md:
```diff
@@ -3,46 +3,17 @@
 # Introduction
 
-Matroska aims to become THE standard of multimedia container formats. It was derived from a project called [MCF](http://mukoli.free.fr/mcf/mcf.html), but differentiates from it significantly because it is based on [EBML](https://github.com/Matroska-Org/ebml-specification/blob/master/specification.markdown) (Extensible Binary Meta Language), a binary derivative of XML. EBML enables significant adv ...
+Matroska aims to become THE standard of multimedia container formats. It can store timestamped multimedia data but also chapters and tags. The tags add important metadata to indentify and classify the information found in a Matroska Segment. It can tag a whole Segment, separate Tracks, individual Chapters or Attachments.
```
Either backtick element names "Segment", "Tracks", "Chapters", "Attachments", or write them in lower-case.

mbunkus commented on index_tags.md:
```diff
@@ -3,46 +3,17 @@
-The table found below is now generated from the "source" of the Matroska specification. This [XML file](https://github.com/Matroska-Org/foundation-source/blob/master/spectool/specdata.xml) is also used to generate the semantic data used in libmatroska and libmatroska2\. We encourage anyone to use and monitor its changes so your code is spec-proof and always up to date.
-
-Note that versions 1, 2 and 3 have been finalized. Version 4 is currently work in progress. There MAY be further additions to v4.
+This document is a work-in-progress specification defining the Matroska file format as part of the [IETF Cellar working group](https://datatracker.ietf.org/wg/cellar/charter/). It uses basic elements and concept already defined in the Matroska specifications defined by this workgroup.
```
Plural, please: "concepts"

mbunkus commented on index_tags.md:
```diff
@@ -3,46 +3,17 @@
-
-- Storage of a arbitrary and potentially executable data within an `Attachment Element`. `Matroska Readers` that extract or use data from Matroska Attachments SHOULD check that the data adheres to expectations.
-- A `Matroska Attachment` with an inaccurate mime-type.
+Tag values can be either strings or binary blobs. Matroska Codecs inherits security considerations from EBML and Matroska.
```
This document is about Tags, not Codecs.

ablwr commented on index_codec.md:
```diff
@@ -0,0 +1,61 @@
+
+# Introduction
+
+Matroska aims to become THE standard of multimedia container formats. It stores interleaved and timestamped audio/video/subtitle data using various codec. To interpret the codec data, a mapping between the way the data are stored in Matroska and how they are understood by such codec is necessary.
```
Is codec the plural of codec, or should it be `data using various codecs`?

ablwr commented on index_codec.md:
```diff
@@ -0,0 +1,61 @@
+
+# Security Considerations
+
+Matroska Codecs inherits security considerations from EBML and Matroska.
```
`Codecs inherits` -> `Codecs inherit`

ablwr commented on index_matroska.md:
```diff
@@ -187,7 +223,7 @@ In this case only the number of frames in the lace is saved, the size of each fr
 #### SimpleBlock Structure
 
-The SimpleBlock is very inspired by the Block structure. The main differences are the added Keyframe flag and Discardable flag. Otherwise everything is the same.
+The SimpleBlock is very inspired by the [Block structure](({{site.baseurl}}/index.html#block-structure). The main differences are the added Keyframe flag and Discardable flag. Otherwise everything is the same.
```
We can skip out on the site.baseurl bit and just go with #block-structure, which works for the RFC.

ablwr commented on tagging-video-example.md:
```diff
@@ -38,7 +30,7 @@ Here is a very basic example of how you would add the title, director, date of r
 ## Bigger example
 
-Let's consider [the Simpsons DVDs](http://www2.foxstore.com/detail.html?item=361&u=987698787). You have a set of DVDs, each containing a few episodes of a whole season. The episodes would be spanned as follows :
+Let's consider [the Simpsons DVDs](https://en.wikipedia.org/wiki/List_of_The_Simpsons_home_video_releases). You have a set of DVDs, each containing a few episodes of a whole season. The episodes would be spanned as follows :
```
Is it a goal to avoid using "you" here, to make things for specification-sounding?

retokromer replied:
no space before `:`

mbunkus commented:
As @ablwr is reworking more and more parts of the spec, we should probably decide what to do with this split sooner than later. Both @ablwr and myself have requested changes but in general approved such a split.

Is there anything that's blocking this split in principal (apart from the requested changes)?

JeromeMartinez commented:
> Is there anything that's blocking this split in principal (apart from the requested changes)?

I am in favor of the split.

ablwr commented:
Agreed; my requested changes are small.

robUx4 commented:
I rebased and updated the link to Block Structure. The rest was just typos and bad wording that was already in the previous versions of the spec. This is not what this patch is about.

ablwr commented:
Thank you and congrats!!! I will send over a new PR fixing the small typo issues raised by @mbunkus and myself.
